### PR TITLE
Added allow-empty-catch option to no-empty

### LINKED ISSUE #1088: Request: no-empty should ignore empty catch blocks
I'm getting a `no-empty` error on this:

``` js
try {
  doSomethingThatMayFail();
} catch (e) {
}
```

but not this:

``` js
try {
  doSomethingThatMayFail();
} catch (e) {
  // ignore this
}
```

Of course it's open to interpretation, but I would expect the first example to also be okay. What do you think?

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
#### PR checklist

- [x] Addresses an existing issue: #1088
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [x] Documentation update

#### Overview of change:

Allows catch blocks to be empty.
Fixes #1088.

#### CHANGELOG.md entry:

[enhancement] Added allow-empty-catch option to `no-empty`

### CODE REVIEW

File: src/rules/noEmptyRule.ts [hunk 1]
```diff
@@ -59,6 +71,16 @@ function walk(ctx: Lint.WalkContext<void>) {
     });
 }
 
+function isExcluded(node: ts.Node, options: Options): boolean {
+    const { parent } = node;
+    if (parent === undefined) {
```
Comment (ajafff) [inline src/rules/noEmptyRule.ts]: `parent` will never be undefined here

File: src/rules/noEmptyRule.ts [hunk 2]
```diff
@@ -59,6 +71,16 @@ function walk(ctx: Lint.WalkContext<void>) {
...
+        return false;
+    }
+
+    return isExcludedConstructor(parent)
+        || (options.allowEmptyCatch === true && parent.kind === ts.SyntaxKind.CatchClause);
```
Comment (ajafff) [inline src/rules/noEmptyRule.ts]: you can simply rename `isExcludedConstructor` and add this condition to its return statement

File: src/rules/noEmptyRule.ts [hunk 3]
```diff
@@ -20,16 +20,26 @@ import * as ts from "typescript";
 
 import * as Lint from "../index";
 
+const ALLOW_EMPTY_CATCH = "allow-empty-catch";
+
+interface Options {
+    allowEmptyCatch?: boolean;
```
Comment (ajafff) [inline src/rules/noEmptyRule.ts]: doesn't need to be optional

File: src/rules/noEmptyRule.ts (hunk 2 above)
Comment (JoshuaKGoldberg) [inline src/rules/noEmptyRule.ts]: Ehh, I actually prefer the explicitness. But sure.

**Pull request merged**